server/app/services/applications/ApplicationService.java

```diff
@@ -0,0 +1,49 @@
+package services.applications;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import play.libs.concurrent.HttpExecutionContext;
+import repository.ApplicationRepository;
+
+/** The service responsible for mediating access to the Application resource. */
+public final class ApplicationService {
+
+  private final ApplicationRepository applicationRepository;
+  private final HttpExecutionContext httpExecutionContext;
+
+  @Inject
+  ApplicationService(
+      ApplicationRepository applicationRepository, HttpExecutionContext httpExecutionContext) {
+    this.applicationRepository = checkNotNull(applicationRepository);
+    this.httpExecutionContext = checkNotNull(httpExecutionContext);
+  }
+
+  /** Validates that the given application is not empty or without a program name. */
+  public Optional<Application> validateApplication(Optional<Application> application) {
+    if (application.isEmpty() || application.get().getProgramName().isEmpty()) {
+      return Optional.empty();
```

Comment:
**suggestion (blocking):** AFAIK the system never creates applications that are not associated with a program, I don't think this method is necessary?